clang/lib/Driver/ToolChains/Clang.cpp

```diff
@@ -6653,12 +6651,15 @@ void Clang::ConstructJob(Compilation &C, const JobAction &JA,
         }
       }
     }
-    if (IsSYCLOffloadDevice && JA.getType() == types::TY_SYCL_Header) {
-      // Generating a SYCL Header
-      SmallString<128> HeaderOpt("-fsycl-int-header=");
-      HeaderOpt += Output.getFilename();
-      CmdArgs.push_back(Args.MakeArgString(HeaderOpt));
+    if (IsSYCLOffloadDevice) {
```

Comment:
will this be fine if you use `-save-temps` ?

Reply:
The header file will be retained, but the name of the header will still be a temporary name and in the /tmp directory.

Reply:
The temporary integration file will be removed during regular non `-save-temps` compilations.  It will not be removed when `-save-temps` is used.  The nuance here is typically when using `-save-temps` the temporary files are not generated in the `TMP` directory and will be named based on expected target and extension and saved in `$CWD`